mod/search/start.php

```diff
@@ -495,6 +686,49 @@ function search_output_tag(\Elgg\Hook $hook) {
 	return $vars;
 }
 
+/**
+ * Format comment entity in search results
+ *
+ * @elgg_plugin_hook search:format entity
+ *
+ * @param \Elgg\Hook $hook Hook
+ * @return \ElggEntity
+ */
+function search_format_comment_entity(\Elgg\Hook $hook) {
```

Comment:
can't we move this to a view?

Reply:
I want to stop doing crazy shit in views

Reply:
instead do crazy shit in functions :) not going to block on this

Reply:
A function can be tested, view testability is a hassle

Reply:
In a perfect world, views are templates, they should not be altering entities, setting/resetting global state, especially without cleaning up after themselves

Reply:
agree.... only wondering why this has to happen in the search plugin and not somewhere else

Reply:
Because search plugin is the one using the volatile data

Reply:
ah, maybe the word `format` is a bit confusing as it only prepares volatile data on the entity...